CC/Sounder/BaseRadioSet.cc

```diff
@@ -173,45 +173,48 @@ BaseRadioSet::BaseRadioSet(Config* cfg)
+                    dev->writeSetting("TDD_MODE", "true");
+                    dev->writeSetting("TDD_CONFIG", tddConfStr);
+                }
 
-            // write beacons to FPGA buffers
-            size_t ndx = 0;
-            for (size_t i = 0; i < bsRadios[0].size(); i++) {
-                SoapySDR::Device* dev = bsRadios[0][i]->dev;
-                dev->writeRegisters("BEACON_RAM", 0, _cfg->beacon);
-                for (char const& c : _cfg->bsChannel) {
-                    bool isBeaconAntenna = !_cfg->beamsweep && ndx == _cfg->beacon_ant;
-                    std::vector<unsigned> beacon_weights(nBsAntennas[0], isBeaconAntenna ? 1 : 0);
-                    std::string tx_ram_wgt = "BEACON_RAM_WGT_";
-                    if (_cfg->beamsweep) {
-                        for (int j = 0; j < nBsAntennas[0]; j++)
-                            beacon_weights[j] = CommsLib::hadamard2(ndx, j);
+                // write beacons to FPGA buffers
+                size_t ndx = 0;
+                for (size_t i = 0; i < bsRadios[c].size(); i++) {
+                    SoapySDR::Device* dev = bsRadios[c][i]->dev;
+                    dev->writeRegisters("BEACON_RAM", 0, _cfg->beacon);
+                    for (char const& ch : _cfg->bsChannel) {
+                        bool isBeaconAntenna = !_cfg->beamsweep && ndx == _cfg->beacon_ant;
+                        std::vector<unsigned> beacon_weights(nBsAntennas[c], isBeaconAntenna ? 1 : 0);
+                        std::string tx_ram_wgt = "BEACON_RAM_WGT_";
+                        if (_cfg->beamsweep) {
+                            for (int j = 0; j < nBsAntennas[c]; j++)
+                                beacon_weights[j] = CommsLib::hadamard2(ndx, j);
+                        }
+                        dev->writeRegisters(tx_ram_wgt + ch, 0, beacon_weights);
+                        ++ndx;
                     }
-                    dev->writeRegisters(tx_ram_wgt + c, 0, beacon_weights);
-                    ++ndx;
+                    dev->writeSetting("BEACON_START", std::to_string(bsRadios[c].size()));
                 }
-                dev->writeSetting("BEACON_START", std::to_string(bsRadios[0].size()));
             }
-        }
 
-        for (size_t i = 0; i < bsRadios[0].size(); i++) {
-            SoapySDR::Device* dev = bsRadios[0][i]->dev;
-            if (!kUseUHD) {
-                bsRadios[0][i]->activateRecv();
-                bsRadios[0][i]->activateXmit();
-                dev->setHardwareTime(0, "TRIGGER");
-            } else {
-                dev->setHardwareTime(0, "UNKNOWN_PPS"); // "CMD"
-                bsRadios[0][i]->activateRecv();
-                bsRadios[0][i]->activateXmit();
+            for (size_t i = 0; i < bsRadios[c].size(); i++) {
+                SoapySDR::Device* dev = bsRadios[c][i]->dev;
+                if (!kUseUHD) {
+                    bsRadios[c][i]->activateRecv();
+                    bsRadios[c][i]->activateXmit();
```

Comment:
Can `dev` be used in these two lines?

Reply:
not sure, not my code but I don't think so.